library/src/main/java/com/novoda/downloadmanager/AndroidLocalFilesDirectory.java

```diff
@@ -15,7 +17,15 @@
 
     @Override
     public List<String> contents() {
-        return Arrays.asList(context.getFilesDir().list());
+        File filesDir = context.getFilesDir();
+        if (filesDir != null) {
+            String[] directoryAsStrings = filesDir.list();
+            if (directoryAsStrings != null) {
+                return Arrays.asList(directoryAsStrings);
```

Comment:
Possible null pointer dereference when applying just `Arrays.asList(filesDir.list())` as `filesDir.list()` can potentially return a null and `Arrays.asList` expects `non-null` values.

https://sbforge.org/sonar/rules/show/findbugs:NP_NULL_ON_SOME_PATH?layout=false